pyrobuf/parse_proto.py

```diff
@@ -124,68 +156,17 @@ def tokenize(self):
             subm = self.token_getter[token_type](m.group(token_type))
             vals = subm.groups()
 
-            if token_type == 'OPTION':
-                yield self.Option(line, *vals)
-
-            elif token_type == 'SYNTAX':
-                yield self.Syntax(line, *vals)
-
-            elif token_type == 'IMPORT':
-                yield self.Import(line, *vals)
-
-            elif token_type == 'MESSAGE':
-                yield self.Message(line, *vals)
-
-            elif token_type == 'MODIFIER':
-                yield self.Modifier(line, *vals)
-
-            elif token_type == 'FIELD':
-                yield self.Field(line, *vals)
-
-            elif token_type == 'LBRACKET':
-                yield self.LBracket(line)
-
-            elif token_type == 'RBRACKET':
-                yield self.RBracket(line)
-
-            elif token_type == 'DEFAULT':
-                yield self.Default(line, *vals)
+            assert token_type not in disabled_token_types, (
```

Comment:
I think we may just want to throw a warning here, or make failing or warning an option.

Reply:
Currently the map field and oneof where just failing the code without good explanation. I do worry that if we go for warnings and skip the assertions, people we'll get weird exceptions in the code that make the C extensions and it would be hard to trace the origin.

Since there will be exceptions down the line anyway, I preferred to raise them where it would make sense and the user would sure understand why things are not working. What do you think?

Reply:
Okay, that makes sense.